### PR TITLE
[kube-prometheus-stack] Add downward compat for Prom CRD

### PR DESCRIPTION
**Pull request by schnatterer**
#### What this PR does / why we need it

Running the latest prom-stack versions on legacy OpenShift clusters with no influence on the preinstalled CRDs results in errors such as this:

```
failed to create typed patch object (..): .spec.automountServiceAccountToken: field not declared in schema
```

This PR provides a workaround using this values.yaml:

```
prometheus:
  prometheusSpec:
    automountServiceAccountToken: null
```

Yes, this is no ideal solution, but that seems to be what the enterprise-world  requires 😐️

#### Special notes for your reviewer

This is the same as #4906 only for the new field 
`prometheus.prometheusSpec.automountServiceAccountToken` that was introduced to the Prometheus CRD in 34b499cf2a368cab94d5c1b1a7784a7b6cef88df and then added to the chart in ff10663d1c9fe327afccbf70698649621761079e, which leads to the incompatibility.

@jkroepke Could I ask you again to review this?

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -17,7 +17,9 @@ metadata:
 {{ toYaml .Values.prometheus.annotations | indent 4 }}
 {{- end }}
 spec:
+{{- if not (eq .Values.prometheus.prometheusSpec.automountServiceAccountToken nil) }}
```
Comment (jkroepke) [inline charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml:20]: If I remember from last PR, this eq nil check require a minimum helm version?

Comment (schnatterer) [reply charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml]: You're right! #4976 was the one about needing Helm 3.11.
I only remembered the subchart issue in #4818 😬
Compatibility issues everywhere!

Fixed in 7d8c6d1c.

**Comment by jkroepke** [comment]
Seems like dashboards needs to be generated first.

**Review by jkroepke — APPROVED** [review]
LGTM

**Pull request merged**